ckan/config/middleware/flask_app.py

```diff
@@ -561,8 +575,21 @@ def _setup_webassets(app: CKANApp):
 
     webassets_folder = get_webassets_path()
 
-    def webassets(path: str):
-        return send_from_directory(webassets_folder, path)
+    def webassets(path: str) -> Response:
+        request.environ['__webasset__'] = True
+
+        cache_expire = config.get(u'ckan.cache_expires', 0)
+        if cache_expire == 0:
+            # If set, ``Cache-Control`` will be ``public``, otherwise
+            #         it will be ``no-cache``
+            cache_expire = None
+        response = send_from_directory(webassets_folder, path,
+                                       etag=True, max_age=cache_expire)
+
+        shared_cache_expire = config.get(u'ckan.shared_cache_expires', 0)
+        if cache_expire is not None and cache_expire != 9999999:
```

Comment:
where does `9999999` come from?

Reply:
@smotornyuk do we have a common way to allow different types for config options, or would we need to use a separate option for this?